embedded/store/indexer.go

```diff
@@ -77,30 +85,71 @@ var (
 	})
 )
 
-func newIndexer(path string, store *ImmuStore, indexOpts *tbtree.Options, maxWaitees int) (*indexer, error) {
+func newIndexer(path string, store *ImmuStore, opts *Options) (*indexer, error) {
+	if store == nil {
+		return nil, fmt.Errorf("%w: nil store", ErrIllegalArguments)
+	}
+
+	indexOpts := tbtree.DefaultOptions().
+		WithReadOnly(opts.ReadOnly).
+		WithFileMode(opts.FileMode).
+		WithLogger(opts.logger).
+		WithFileSize(opts.FileSize).
+		WithCacheSize(opts.IndexOpts.CacheSize).
+		WithFlushThld(opts.IndexOpts.FlushThld).
+		WithSyncThld(opts.IndexOpts.SyncThld).
+		WithFlushBufferSize(opts.IndexOpts.FlushBufferSize).
+		WithCleanupPercentage(opts.IndexOpts.CleanupPercentage).
+		WithMaxActiveSnapshots(opts.IndexOpts.MaxActiveSnapshots).
+		WithMaxNodeSize(opts.IndexOpts.MaxNodeSize).
+		WithMaxKeySize(opts.MaxKeyLen).
+		WithMaxValueSize(lszSize + offsetSize + sha256.Size + sszSize + maxTxMetadataLen + sszSize + maxKVMetadataLen). // indexed values
+		WithNodesLogMaxOpenedFiles(opts.IndexOpts.NodesLogMaxOpenedFiles).
+		WithHistoryLogMaxOpenedFiles(opts.IndexOpts.HistoryLogMaxOpenedFiles).
+		WithCommitLogMaxOpenedFiles(opts.IndexOpts.CommitLogMaxOpenedFiles).
+		WithRenewSnapRootAfter(opts.IndexOpts.RenewSnapRootAfter).
+		WithCompactionThld(opts.IndexOpts.CompactionThld).
+		WithDelayDuringCompaction(opts.IndexOpts.DelayDuringCompaction)
+
```

Comment:
I think the part to validate indexopts is missing? 
This line I mean err = indexOpts.Validate()

Reply:
actually, the opts are validated by the btree itself